File: xkcd/src/main/java/xyz/jienan/xkcd/settings/PreferenceActivity.java
package xyz.jienan.xkcd.settings;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import android.view.MenuItem;

import xyz.jienan.xkcd.base.BaseActivity;

/**
 * Created by Jienan on 2018/3/9.
 */

public class PreferenceActivity extends BaseActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getFragmentManager().beginTransaction()
                .replace(android.R.id.content, new SettingsFragment()).commit();
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                break;
        }
        return false;
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/settings/PreferenceActivity.kt
package xyz.jienan.xkcd.settings

import android.os.Bundle
import android.view.MenuItem
import xyz.jienan.xkcd.base.BaseActivity

/**
 * Created by Jienan on 2018/3/9.
 */

class PreferenceActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportFragmentManager.beginTransaction()
                .replace(android.R.id.content, SettingsFragment()).commit()
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> finish()
        }
        return false
    }
}
